### PR TITLE
#139 - bubbles generator

### LINKED ISSUE #139: Bubbles generator
I would like to have a `/bubbles` or whatever page without navigation, content and footer. 

Sometimes we need to get background with bubbles like here:
obraz

Maybe some configurator/randomizer would be nice too.

### PR DESCRIPTION
**Pull request by kamilpiech97**
This should close #139 .
Added **/bubbles/[numberOfBubbles]** page without navigation and footer. Link for that page is as an easter egg. We can pass into link number of needed bubbles on page.

**Page with generated bubbles:**
obraz

### CODE REVIEW

File: src/layout/bubbles.svelte [hunk 1]
```diff
@@ -25,6 +25,11 @@
     { left: 92, size: 108, delay: 0, duration: 8, popping: false }
   ]
 
+  if (window.location.pathname === '/bubbles') {
```
Comment (krzysztofrewak) [inline src/layout/bubbles.svelte:28]: Oh. With this approach you are checking if path is bubbles on every single page. I think it's a overkill here.

Maybe just create a new route/page and disable inherited layout somehow?

File: src/layout/bubbles.svelte [hunk 2]
```diff
@@ -25,6 +25,11 @@
...
+    const times = prompt('Please enter number of bubbles (10 - 200):', 10)
```
Comment (krzysztofrewak) [inline src/layout/bubbles.svelte:29]: I would prefer passing this parameter in URL. Like /bubbles/50.

File: src/layout/bubbles.svelte [hunk 3]
```diff
@@ -88,6 +93,27 @@
...
+    const generatedBubbles = []
+    for (let i = 0; i < times; i++) {
+      generatedBubbles.push(
+        { left: getRndInteger(1, 99), size: getRndInteger(20, 150), delay: getRndInteger(4, 10), duration: getRndInteger(5, 14), popping: false }
+      )
+    }
+    return generatedBubbles
+  }
+
+  function getRndInteger (min, max) {
```
Comment (krzysztofrewak) [inline src/layout/bubbles.svelte:107]:
```suggestion
  function getRandomInteger (min, max) {
```
C'mon.

File: src/layout/footer.svelte [hunk 4]
```diff
@@ -15,6 +15,7 @@
                     <a href="{ $url('./privacy') }" class="hover:text-gray-900 ml-4">{ $_('footer.privacy') }</a>
                     <a href="{ $url('./company') }" class="hover:text-gray-900 ml-4">{ $_('footer.company') }</a>
                     <a href="https://github.com/blumilksoftware/website" target="_blank" class="hover:text-gray-900 ml-4">{ $_('footer.source') }</a>
+                    <a href="{ $url('./bubbles') }" target="_blank" class="hover:text-gray-900 ml-4">{ $_('footer.bubbles') }</a>
```
Comment (krzysztofrewak) [inline src/layout/footer.svelte:18]: I wouldn't share this page for public like this. Let's keep it an easter egg.

File: src/pages/bubbles.svelte [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<svelte:head>
```
Comment (krzysztofrewak) [inline src/pages/bubbles.svelte:1]: Oh, there's a page for that. Move bubbles logic here.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: src/pages/bubbles.svelte (hunk 5 above)
Comment (kamilpiech97) [reply src/pages/bubbles.svelte]: moved logic into bubble file

File: src/layout/footer.svelte (hunk 4 above)
Comment (kamilpiech97) [reply src/layout/footer.svelte]: changed.

File: src/layout/bubbles.svelte (hunk 3 above)
Comment (kamilpiech97) [reply src/layout/bubbles.svelte]: changed.

File: src/layout/bubbles.svelte (hunk 2 above)
Comment (kamilpiech97) [reply src/layout/bubbles.svelte]: changed.

File: src/layout/bubbles.svelte (hunk 1 above)
Comment (kamilpiech97) [reply src/layout/bubbles.svelte]: I moved all whole new and old bubbles logic into separated file - first file with bubbles stay without any changes.

File: .gitignore [hunk 6]
```diff
@@ -2,3 +2,4 @@
 /node_modules/
 /public/build/
 .env
+/cypress/
```
[the remark this answers is not in the document]
Comment (kamilpiech97) [reply .gitignore]: I deleted that.

File: src/pages/bubbles/[number].svelte [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    function pop (event, index) {
+      if (!bubbles[index].popping) {
+        bubbles[index].popping = true
+        for (let i = 0; i < Math.random() * 10 + 2; i++) {
+          createParticle(event.target.parentElement.getElementsByClassName('particles')[0])
+        }
+      }
+    }
+
+    function createParticle (parent) {
```
Comment (krzysztofrewak) [inline src/pages/bubbles/[number].svelte:41]: Do we need this code copypasted from `src/layout/bubbles.svelte`?

File: src/pages/bubbles/[number].svelte [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+             style="{ getStyle(bubble) }"
+             class:popping="{ bubbles[index].popping }"
+             on:animationiteration={ (event) => animationEndHandler(event, index) }
+        >
+            <div class="inner"></div>
+            <div class="particles"></div>
+        </div>
+    {/each}
+</div>
+<script>
```
Comment (krzysztofrewak) [inline src/pages/bubbles/[number].svelte:14]:
```suggestion
</div>

<script>
```
I would separate these.

File: src/pages/bubbles/[number].svelte [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    function checkTimes (times) {
+      if (times > 0 && times < 200 && (typeof times === 'number')) {
+        return times
+      }
+      return 15
+    }
+</script>
+<style>
```
Comment (krzysztofrewak) [inline src/pages/bubbles/[number].svelte:107]:
```suggestion
</script>

<style>
```

File: src/pages/_layout.svelte [hunk 10]
```diff
@@ -7,10 +8,11 @@
 <Footer items={items} socials={socials} contact={contact}/>
 
 <script>
-  import items from '@/assets/navigation.js'
-  import socials from '@/assets/socials.js'
-  import contact from '@/assets/contact.js'
+    import items from '@/assets/navigation.js'
```
Comment (krzysztofrewak) [inline src/pages/_layout.svelte:11]: Why did you add these indents here?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: src/pages/_layout.svelte (hunk 10 above)
Comment (kamilpiech97) [reply src/pages/_layout.svelte:11]: My IDE do things like this. Do you have custom style for svelte (your style)?

Comment (krzysztofrewak) [reply src/pages/_layout.svelte:11]: Hm, I think I have just a Svelte plugin: https://plugins.jetbrains.com/plugin/12375-svelte

File: src/pages/bubbles/[number].svelte (hunk 7 above)
Comment (kamilpiech97) [reply src/pages/bubbles/[number].svelte]: Changed, take a look and give me feedback about my solution.

File: src/pages/_layout.svelte (hunk 10 above)
Comment (kamilpiech97) [reply src/pages/_layout.svelte]: I think, now all is okay :)

**Review by krzysztofrewak — APPROVED** [review]

**Review by jsawo — APPROVED** [review]
